main.go

```diff
@@ -336,17 +339,28 @@ func runCmpBenches(
 	runPattern, benchTime string,
 	cpuProfile, memProfile, mutexProfile bool,
 	itersPerTest int,
+	preview bool,
 ) error {
 	var spinner ui.Spinner
-	spinner.Start(os.Stderr, "running benchmarks:")
+	spinner.Start(os.Stderr, "running benchmarks:\n")
 	defer spinner.Stop()
 	for i, t := range tests {
 		pkg := testBinToPkg(t)
 		for j := 0; j < itersPerTest; j++ {
 			pkgFrac := ui.Fraction(i+1, len(tests))
 			iterFrac := ui.Fraction(j+1, itersPerTest)
-			progress := fmt.Sprintf(" pkg=%s iter=%s %s", pkgFrac, iterFrac, pkg)
-			spinner.Update(progress)
+			var buf bytes.Buffer
+			if preview {
+				_, err := processBenchOutput(ctx, &buf, bs1, bs2, true, text, tests, nil)
+				if err != nil {
+					return err
+				}
+				_, _ = fmt.Fprintln(&buf)
```

Comment:
[supernit] There's an extra space because of how we're using the spinner. Maybe add a `\n` at the beginning of `buf` so it doesn't matter. Or change the spinner to not add the space if `prefix` is empty and use an empty prefix.